### ISSUE TITLE
"The connection to the server has been lost." when attempting to start game

### ISSUE THREAD

**Issue by rookie10276** [freeorion/freeorion#3110]
v0.4.10 [build 2020-07-10.f3d403e]
Windows 10 Pro 
Nvidia GTX 970
Binary release v0.4.10 RC1

Gameplay screen does not start when selecting Single Player, Quickstart, Multiplayer. Instead, a popup "The connection to the server has been lost." appears in the main menu. This popup also occurs when canceling out of Single Player menu.

Attempted troubleshooting by checking for executables in "Windows Defender Firewall with Advanced Security", "Allow an app through Windows Firewall", and disabling autosave in-game. All of which had no effect.

Steps to reproduce
1. Attempting to start a game from Single Player, Quickstart, or Multiplayer. Or hitting cancel on the Single Player menu.
2. Screen returns to main menu with error popup.

[[debug]freeorion.log](https://github.com/freeorion/freeorion/files/4944522/debug.freeorion.log)
[[debug]freeoriond.log](https://github.com/freeorion/freeorion/files/4944521/debug.freeoriond.log)
[[trace]freeorion.zip](https://github.com/freeorion/freeorion/files/4944526/trace.freeorion.zip)
[[trace]freeoriond.log](https://github.com/freeorion/freeorion/files/4944524/trace.freeoriond.log)

Edit: I found someone with a similar issue 2 years back: https://www.freeorion.org/forum/viewtopic.php?t=11075
Suggestion in forum post is to add command line to change network port. When launching and choosing QuickStart the game will hang for ~10-20 seconds and bring me to the main menu with the popup "Timed out while attempting to connect to server". 
Any attempts with Single Player or Multiplayer or subsequent attempts with Quickstart will give the popup "The connection to the server has been lost" on the main menu

Steps to reproduce:
1. Launching with a command line option "--network.message.port [different port number]" 
2. Clicking QuickStart 

[[debug]freeoriond.log](https://github.com/freeorion/freeorion/files/4946082/debug.freeoriond.log)
[[debug]freeorion.log](https://github.com/freeorion/freeorion/files/4946083/debug.freeorion.log)

**Comment by Oberlus** [comment]
Can't reproduce (Windows 7 and 8.1, Ubuntu 18.04).
Can't see the source of the problem in the logs. Only error is in freeorion.log, access denied in a kill call?

**Comment by geoffthemedio** [comment]
Does it make any difference if you start a single player game with no AIs?

Otherwise, I have no idea... There's nothing about an error in the server log... it just ends in the middle of parsing stuff. The client log just says "Client connection disconnected, due to connection reset from server."

**Comment by rookie10276** [comment]
Attempted to start Single Player game with 0 AI, but just bounced me back to the main screen with "The connection to the server has been lost" popup.

**Comment by geoffthemedio** [comment]
If you click multiplayer, does it go to the multiplayer lobby, or immediately show the error popup?

If you run a command prompt (cmd.exe), navigate to wherever you have FreeOrion installed, and then start `freeoriond.exe` separately, does it sit waiting for network events or immediately abort?

If it sits waiting, try running `freeorion --network.server.external.force 1` (also from the command line) and see what happens?

**Comment by Oberlus** [comment]
Maybe not related, but I once installed the game wrongly due to bad choices in the popups from the Comodo Firewall/Security app and that lead to similar (but not equal) problems, due to restricted permissions for freeoriond.exe.
Try reinstalling?

**Comment by rookie10276** [comment]
> If you click multiplayer, does it go to the multiplayer lobby, or immediately show the error popup?
> 
> If you run a command prompt (cmd.exe), navigate to wherever you have FreeOrion installed, and then start `freeoriond.exe` separately, does it sit waiting for network events or immediately abort?
> 
> If it sits waiting, try running `freeorion --network.server.external.force 1` (also from the command line) and see what happens?

I'm not getting to the lobby. I have the popup of whether to choose join or host. Then it hangs for a short while and brings the usual popup.

I also attempted a full reinstall of the game with no different results. I also tried launching freeorion through the command line with the "--network.server.external.force 1" command line. I hate to say it but i am wonder if it's just some outlier error with my OS in particular.

**Comment by Vezzra** [comment]
> This popup also occurs when canceling out of Single Player menu.

I can cofirm at least that part on macOS and Windows, with the RC1 builds. Apart from that I can start games just fine...

**Comment by Thinker55** [comment]
I don't know if this is the same issue or a different one.  I am 77 turns into a single-player game.  Suddenly, I get "The connection to the server has been lost" message.  Should there be a server connection in single player mode?  Nothing I can do to restart the game or recover from saved file gets me past this.
Environment SuSE LEAP 15.2.

**Comment by geoffthemedio** [comment]
> Should there be a server connection in single player mode?

Yes. The FreeOrion server program runs on your local machine and communicates with the human and AI clients by non-network messaging.

> Nothing I can do to restart the game or recover from saved file gets me past this.

The same info as requested for the original issue would be useful: attach logs from immediately after observing the issue, and and a save file that reproduces it. That said, your issue appears to be with a saved game not loading, whereas this thread was about not being able to start a game. If that's accurate, please make a new bug report.

**Comment by life-on-mars** [comment]
I've been having this issue since 0.4.10 on Windows 10. I tried installing it on a headless w10 machine which is practically a clean install and it runs without problems. 
The message "connection to server is lost" appears with anything where a server connections is made:
- single player
- quickstart
- hosting multiplayer

Connecting to multiplayer times out and hosted games on other machine do not appear in the list. Firewall settings do not restrict freeorion in any way and I tried reverting back to 0.4.5 and 0.4.9 which work flawlessly.
